nuitka/build/static_src/MainProgram.c

```diff
@@ -1367,6 +1379,81 @@ void setLANGSystemLocaleMacOS(void) {
 #endif
 
 static int Nuitka_Main(int argc, native_command_line_argument_t **argv) {
+
+#if(defined(_NUITKA_LINUX_CONSOLE_MODE_FORCE) || defined(_NUITKA_LINUX_CONSOLE_MODE_DETECT)) && !defined(_WIN32) && !defined(__APPLE__)
+
+if (!isatty(STDIN_FILENO)) {
+
+    const * already_relaunched = getenv("NUITKA_TERMINAL_RELAUNCH");
+    if (already_relaunched == NULL) {
+        setenv("NUITKA_TERMINAL_RELAUNCH", "1", 1);
+
+
+        char
+        const * binary_path = getBinaryFilenameHostEncoded(true);
+
+        char ** spawn_argv = (char ** ) malloc(sizeof(char * ) * (argc + 2));
+        if (spawn_argv == NULL) {
+            fprintf(stderr, "Error: Memory allocation failed for xdg-terminal-exec argv.\n");
+            exit(1); // Exit indicating a failure
+        }
+
+
+        spawn_argv[0] = "xdg-terminal-exec";
+        spawn_argv[1] = (char * ) binary_path;
+        for (int i = 1; i < argc; i++) {
+            spawn_argv[i + 1] = argv[i];
+        }
+        spawn_argv[argc + 1] = NULL;
+
+        pid_t pid;
+        extern char ** environ;
+        int spawn_result = posix_spawnp( & pid, "xdg-terminal-exec", NULL, NULL, spawn_argv, environ);
```

Comment:
Is it not possible to use an exec function like `execv` (or other variant), because why keep this around waiting? The new process can do all the error exiting. Of course if the command is not found and not working, it will be ugly, but mind you, we got no away of talking to the user anyway with no terminal.